compiler/hash-utils/src/printing.rs

```diff
@@ -0,0 +1,39 @@
+use std::fmt;
+
+/// Type wrapper around a slice of items.
+pub struct SliceDisplay<'a, T: 'a>(pub &'a [T]);
```

Comment:
Given the docs below, maybe it would be better to call this "SequenceDisplay" or something, not the programming-specific "SliceDisplay". Also since this is the purpose of this struct, maybe the doc comment should be on the struct itself and not on the fmt::Display implementation.